native/acter/src/api/spaces.rs

```diff
@@ -34,19 +35,20 @@ use ruma_events::{
     space::child::SpaceChildEventContent, AnyStateEventContent, MessageLikeEvent, StateEventType,
 };
 use serde::{Deserialize, Serialize};
-use std::ops::Deref;
+use std::{ops::Deref, sync::Arc};
 use tokio::sync::broadcast::Receiver;
 use tokio_stream::{wrappers::BroadcastStream, Stream};
 use tracing::{error, trace, warn};
 
-use crate::{Client, PublicSearchResult, Room, RUNTIME};
+use crate::{Client, PublicSearchResult, Room, TimelineStream, RUNTIME};
 
 use super::utils::{remap_for_diff, ApiVectorDiff};
 
 #[derive(Debug, Clone)]
 pub struct Space {
     pub client: Client,
     pub(crate) inner: Room,
+    timeline: Arc<Timeline>,
```

Comment:
Nope. Similar as for chat-room, we currently _don't_ ensure that `Space` is globally unique for the same `roomId`. It is created and re-created and cloned multiple times in the system and created ad-hoc. Storing a stateful object on it is not a suitable way to solve whatever problem this is meant to solve.

Implying that it was and there the `Timeline` would a unique and always alive object is not reasonable. Other from the fact that keeping a Timeline object around all the time for each space is too costly. We do not usually care about the _timeline_-based order for events in a space.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1267/commits/029bdeae47415c616680404bdcb3e9454238fd1f